test/distributed_util.py

```diff
@@ -93,11 +90,13 @@ def train_step(model, inputs, labels, optimizer, loss_fn):
   return loss
 
 
-def ddp_correctness(init_file: str,
-                    *,
+def ddp_correctness(ddp: type = torch.nn.parallel.DistributedDataParallel,
                     use_large_net: bool = False,
                     debug: bool = False):
-  rank, world_size = init_xla_backend(init_file)
+  if not dist.is_initialized():
+    rank, world_size = init_xla_backend()
+  else:
+    rank, world_size = xm.get_ordinal(), xm.xrt_world_size()
```

Comment:
one customer asked if they can use `torch.distributed` to get the ordinal and world size, do you know if that will work if we init the xla backend?

Reply:
if I understand correctly
1. if user use the upstream ddp, they need to manually `init_process_group`
2. if user use our custom wrapper, they don't.

If this a fair understanding?

Reply:
Ordinal and world size will be correct in `torch.distributed` on TPU v4 if the process group is initialized with `init_pjrt_process_group`, but not on v2/v3, because we'll have two ordinals per rank on v2/v3.

I added the option to initialize the torch.distributed group through `xmp.spawn` for three reasons, in order of importance:

1. The master IP is not predictable in a TPU slice.
2. If on TPU v2/v3, only one thread can initialize the `torch.distributed` group. It's easier to do it here than tell the user to manage that themselves.
3. It may be convenient for the PJRT process ID to match each process' rank so we have one fewer index to keep track of.

So, with either DDP implementation, the user can initialize the main process group themselves if they solve the three problems above.

The default is to not initialize the process group since we found that TCP process group initialization is flaky on large pod slices.